src/server/main.py

```diff
@@ -1,30 +1,345 @@
-from flask import Flask
+import json
+import uuid
+from datetime import datetime, timedelta
+from functools import wraps
+from sqlite3 import Connection, Cursor
+from sqlite3 import Error as SqlError
+from typing import Any, Callable
+from uuid import UUID
+
+import jwt  # type: ignore
+from flask import Flask, Response, make_response, request  # type: ignore
+from werkzeug.security import (  # type: ignore
+    check_password_hash,
+    generate_password_hash,
+)
+
+from .database_provider import getDatabaseConnection
 
 app = Flask(__name__)
+# Generetion method -> uuid.uuid4().hex
+app.config["SECRET_KEY"] = "55cfba6d5bd6405c8e9b7b681f6b8835"
 
 
-@app.route("/me")
-def me() -> str:
-    """User's information endpoint."""
-    return "me"
+def token_required(fun: Callable[..., Response]) -> Callable[..., Response]:
+    """Validate received token."""
+
+    @wraps(fun)
+    def decorated(*args: tuple, **kwargs: dict) -> Response:
+        """Validate received token."""
+        token: str = ""
+        # JWT is passed in the request header
+        if "x-access-token" in request.headers:
+            token = request.headers["x-access-token"]
+
+        # If the token is not passed return message and exit function
+        if not token:
+            return make_response(
+                json.dumps({"message": "Token is missing"}),
+                401,
+                {"WWW-Authenticate": 'Basic realm ="Token is missing!"'},
+            )
+
+        # If the token is revoked return message and exit function
+        if is_token_revoked(token):
+            return make_response(
+                json.dumps({"message": "Token is revoked"}),
+                401,
+                {"WWW-Authenticate": 'Basic realm ="Token is revoked!"'},
+            )
+
+        # Decode the token and retrieve the information contained in it
+        try:
+            data: dict[str, Any] = jwt.decode(token, app.config["SECRET_KEY"], algorithms=["HS256"])
+        except jwt.InvalidTokenError as e:
+            return make_response(
+                json.dumps({"message": f"Token is invalid: {e}"}),
+                401,
+                {"WWW-Authenticate": 'Basic realm ="Token is invalid!"'},
+            )
```

Comment:
According to the [https://flask.palletsprojects.com/en/2.2.x/api/#flask.Flask.make_response](documentation) any `dict` passed to `make_response` will have already been serialized to JSON string. I think that calling `json.dumps` is not necessary here, but feel free to double-check it yourself.

Reply:
Yes, you are right. I've replaced 'json.dumps' with dictionaries in every 'make_response' method.